src/main/java/bio/terra/workspace/service/resource/controlled/ControlledAiNotebookInstanceResource.java

```diff
@@ -0,0 +1,196 @@
+package bio.terra.workspace.service.resource.controlled;
+
+import bio.terra.workspace.common.exception.InconsistentFieldsException;
+import bio.terra.workspace.common.exception.MissingRequiredFieldException;
+import bio.terra.workspace.db.DbSerDes;
+import bio.terra.workspace.db.model.DbResource;
+import bio.terra.workspace.service.resource.ValidationUtils;
+import bio.terra.workspace.service.resource.WsmResourceType;
+import bio.terra.workspace.service.resource.model.CloningInstructions;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.UUID;
+import javax.annotation.Nullable;
+
+/** A {@link ControlledResource} for a Google AI Platform Notebook instance. */
+public class ControlledAiNotebookInstanceResource extends ControlledResource {
+  private final String instanceId;
+  private final String location;
+
+  @JsonCreator
+  public ControlledAiNotebookInstanceResource(
+      @JsonProperty("workspaceId") UUID workspaceId,
+      @JsonProperty("resourceId") UUID resourceId,
+      @JsonProperty("name") String name,
```

Comment:
it's nice to have a prefix to distinguish the "resource name" in the resource table from the, er, name of the resource (in the cloud).